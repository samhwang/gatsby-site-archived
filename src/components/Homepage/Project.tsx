import React from 'react';

function Project(props: ProjectProps) {
  const { title, description, url } = props;

  let titleText = <strong>{`${title}:`}</strong>;

  titleText = url ? (
    <a href={url}>
      <strong>{titleText}</strong>
    </a>
  ) : (
    titleText
  );

  return (
    <li>
      <i className="fa-li fa fa-tasks text-warning" />
      <p>{`${titleText} ${description}`}</p>
    </li>
  );
}

export default Project;
