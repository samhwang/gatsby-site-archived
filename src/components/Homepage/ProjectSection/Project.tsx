import { OutboundLink } from 'gatsby-plugin-google-gtag';

export interface ProjectProps {
  title: string;
  description: string;
  url?: string;
}

function Project({ title, description, url }: ProjectProps) {
  const titleText = url ? (
    <OutboundLink href={url}>
      <strong>{`${title}:`}</strong>
    </OutboundLink>
  ) : (
    <strong>{`${title}:`}</strong>
  );

  return (
    <li>
      <i className="fa-li fa fa-tasks text-warning" />
      <p>
        {titleText}
        {` ${description}`}
      </p>
    </li>
  );
}

Project.defaultProps = {
  url: '',
};

export default Project;
