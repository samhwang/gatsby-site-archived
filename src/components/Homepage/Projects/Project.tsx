import React from 'react';

function Project(props: ProjectProps) {
  const { title, description, url } = props;
  const titleText = url ? (
    <a href={url}>
      <strong>{`${title}:`}</strong>
    </a>
  ) : (
    <strong>{`${title}:`}</strong>
  );

  return (
    <li>
      <i className="fa-li fa fa-tasks text-warning" />
      <p>
        {titleText} {description}
      </p>
    </li>
  );
}

export default Project;
