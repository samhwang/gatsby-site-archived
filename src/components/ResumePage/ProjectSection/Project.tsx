import type { FC } from 'react';
import { OutboundLink } from 'gatsby-plugin-google-gtag';

export interface ProjectProps {
  title: string;
  description: string;
  url?: string;
}

const Project: FC<ProjectProps> = ({ title, description, url }) => {
  const titleText = url ? (
    <OutboundLink href={url}>
      <strong>{`${title}:`}</strong>
    </OutboundLink>
  ) : (
    <strong>{`${title}:`}</strong>
  );

  return (
    <li>
      <i className="fa-li fa fa-tasks text-warning" />
      <p>
        {titleText}
        {` ${description}`}
      </p>
    </li>
  );
};

Project.defaultProps = {
  url: '',
};

export default Project;
