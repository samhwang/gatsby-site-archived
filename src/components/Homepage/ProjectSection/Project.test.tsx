/* eslint-disable import/no-extraneous-dependencies */
import renderer from 'react-test-renderer';
import Project from './Project';

describe('Project Component', () => {
  it('Should render without crashing and match snapshot', () => {
    const project = {
      title: 'title',
      description: 'description',
      url: 'https://samhwang.github.io',
    };
    const tree = renderer
      .create(
        <Project
          title={project.title}
          description={project.description}
          url={project.url}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render project without URL', () => {
    const project = {
      title: 'title',
      description: 'description',
    };
    const tree = renderer
      .create(
        <Project title={project.title} description={project.description} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
