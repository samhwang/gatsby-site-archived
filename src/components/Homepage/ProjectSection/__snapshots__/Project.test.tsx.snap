// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Component Should render project without URL 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li>
        <i
          class="fa-li fa fa-tasks text-warning"
        />
        <p>
          <strong>
            title:
          </strong>
           description
        </p>
      </li>
    </div>
  </body>,
  "container": <div>
    <li>
      <i
        class="fa-li fa fa-tasks text-warning"
      />
      <p>
        <strong>
          title:
        </strong>
         description
      </p>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Project Component Should render without crashing and match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li>
        <i
          class="fa-li fa fa-tasks text-warning"
        />
        <p>
          <a
            href="https://samhwang.github.io"
          >
            <strong>
              title:
            </strong>
          </a>
           description
        </p>
      </li>
    </div>
  </body>,
  "container": <div>
    <li>
      <i
        class="fa-li fa fa-tasks text-warning"
      />
      <p>
        <a
          href="https://samhwang.github.io"
        >
          <strong>
            title:
          </strong>
        </a>
         description
      </p>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
